File: mobile_app/lib/src/core/utils/clinical_note.dart
/// Physios often type session notes quickly in lowercase with no closing
/// punctuation. This is a display polish, not a spelling/grammar fixer: it
/// only capitalizes the first letter and adds a trailing full stop when one
/// is missing, leaving the rest of the text exactly as typed. Mirrors web's
/// `lib/clinical-note.ts`.
String formatClinicalNote(String text) {
  final trimmed = text.trim();
  if (trimmed.isEmpty) return trimmed;
  final capitalized = trimmed[0].toUpperCase() + trimmed.substring(1);
  return RegExp(r'[.!?]$').hasMatch(capitalized) ? capitalized : '$capitalized.';
}

File: lib/clinical-note.ts
// Physios often type session notes quickly in lowercase with no closing
// punctuation (e.g. "initial assessment of foot pain..."). This is a display
// polish, not a spelling/grammar fixer: it only capitalizes the first letter
// and adds a trailing full stop when one is missing, leaving the rest of the
// text — including mid-sentence casing — exactly as typed.
export function formatClinicalNote(text: string): string {
  const trimmed = text.trim();
  if (!trimmed) return trimmed;
  const capitalized = trimmed[0].toUpperCase() + trimmed.slice(1);
  return /[.!?]$/.test(capitalized) ? capitalized : `${capitalized}.`;
}
